google/auth/aio/transport/__init__.py

```diff
@@ -23,3 +23,29 @@
 :mod:`google.auth` to make asynchronous requests. :class:`Response` defines the interface
 for the return value of :class:`Request`.
 """
+
+import abc
+
+
+class Response(metaclass=abc.ABCMeta):
+    """Asynchronous HTTP Response Interface."""
+
+    @abc.abstractmethod
+    async def status_code(self):
+        """int: The HTTP status code."""
+        raise NotImplementedError("status_code must be implemented.")
+
+    @abc.abstractmethod
+    def headers(self):
+        """Mapping[str, str]: The HTTP response headers."""
+        raise NotImplementedError("headers must be implemented.")
+
+    @abc.abstractmethod
+    def content(self):
```

Comment:
Should this be async?

Reply:
The `content` property exposed by `aiohttp` is synchronous. I believe it's synchronous for `httpx` as well but I'll double check.

Do we want to make this async?

Reply:
As per our offline discussion, I've updated this to be async.

Reply:
After putting in some more thought, given that we're declaring `content` as a property, it's expected to be computationally inexpensive. We may also not want to unnecessarily give back control to the event loop.

Therefore, having an async property may not be ideal. Let's discuss this further. Will put this PR on hold until then.

Reply:
We've decided to update content to be a method rather than an attribute so we can keep it as async. We've also updated its return type to be an `AsyncGenerator`.